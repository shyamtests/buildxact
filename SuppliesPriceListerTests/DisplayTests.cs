using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using buildxact_supplies;
using buildxact_supplies.Models;
using Xunit;

namespace SuppliesPriceListerTests
{
    
    [UseReporter(typeof(DiffReporter))]
    public class DisplayTests
    {
        [Fact]
        public void TestWritingToConsole()
        {
            var supplyA = new BuildingSupply
            {
                Id = "4000", Name = "100 x 200 x 20mpa Internal Beam",
                Price = 68.00M
            };
            
            var supplyB = new BuildingSupply
            {
                Id = "7f3c48c4-f8b6-453f-b2fa-83ec31dfa85c", Name = "Bobcat to Dig LM of Strip Footing",
                Price = 800M
            };
            
            var supplyC = new BuildingSupply
            {
                Id = "0a360e10-4e35-4e94-bd80-2e8bd6c749f1", Name = "Under Slab Sand 150mm",
                Price = 77.24M
            };

            var supplies = new[] {supplyA, supplyB, supplyC};
            
            using (var consoleOutput = new ConsoleOutput())
            {
                ConsoleWriter.WriteSuppliesToConsole(supplies);

                Approvals.Verify(consoleOutput.GetOuput());
            }
        }
    }
}
