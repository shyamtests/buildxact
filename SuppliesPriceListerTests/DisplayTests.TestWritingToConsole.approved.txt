4000, 100 x 200 x 20mpa Internal Beam, $68.00

0a360e10-4e35-4e94-bd80-2e8bd6c749f1, Under Slab Sand 150mm, $77.24

7f3c48c4-f8b6-453f-b2fa-83ec31dfa85c, Bobcat to Dig LM of Strip Footing, $800.00

